src/methods/clipping/difference.jl

```diff
@@ -0,0 +1,83 @@
+# #  Difference Polygon Clipping
+export difference
+
+
+"""
+    difference(geom_a, geom_b, ::Type{T}; target::Type{Target})
+
+Return the difference between two geometries as a list of geometries. Return an empty list
+if none are found. The type of the list will be constrained as much as possible given the
+input geometries. Furthermore, the user can provide a `taget` type as a keyword argument and
+a list of target geometries found in the difference will be returned. The user can also
+provide a float type that they would like the points of returned geometries to be. 
+
+## Example 
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+poly1 = GI.Polygon([[[0.0, 0.0], [5.0, 5.0], [10.0, 0.0], [5.0, -5.0], [0.0, 0.0]]])
+poly2 = GI.Polygon([[[3.0, 0.0], [8.0, 5.0], [13.0, 0.0], [8.0, -5.0], [3.0, 0.0]]])
+diff_poly = GO.difference(poly1, poly2; target = GI.PolygonTrait)
+GI.coordinates.(diff_poly)
```

Comment:
`GI.coordinates` is legacy we try not to encourage people to use it

Reply:
What do you replace it with?

Reply:
Honestly Idk 😭

In this case nothing is as simple. But you wouldnt use it in real code because the allocations are horrific (every point allocates a vector...)

I guess its fine for demonstrating the values